tests/test_entrypoint.py

```diff
@@ -0,0 +1,239 @@
+"""
+Testing the logic that extracts the entrypoint data:
+    - CLI entrypoints
+    - GUI entrypoints
+    - miscellaneous entrypoints
+"""
+
+from setuptools_pyproject_migration import WritePyproject
+from setuptools.dist import Distribution
+
+
+# Entry point parsing
+
+
+def test_parse_entrypoint_happypath():
+    """
+    Test we can extract the entry point name and target from a string.
+    """
+    assert WritePyproject._parse_entrypoint("ep=project.module:target") == ("ep", "project.module:target")
+
+
+def test_parse_entrypoint_happypath_whitespace():
+    """
+    Test we strip extraneous whitespace from the strings received.
+    """
+    assert WritePyproject._parse_entrypoint("  ep  =  project.module:target  ") == ("ep", "project.module:target")
+
+
+def test_parse_entrypoint_missing_eq():
+    """
+    Test that a string lacking a '=' is refused.
+    """
+    try:
+        WritePyproject._parse_entrypoint("ep project.module:target")
+        assert False, "Should not have been accepted as valid"
+    except ValueError as e:
+        assert str(e).endswith(" is not of the form 'name = module:function'")
+
+
+# Wrapper function logic
+
+
+def test_generate_entrypoints_none():
+    """
+    Test we get an empty dict if entrypoints is None.
+    """
+    assert WritePyproject._generate_entrypoints(None) == {}
+
+
+def test_generate_entrypoints():
+    """
+    Test we get all entry points grouped by type.
+    """
+    assert WritePyproject._generate_entrypoints(
+        {
+            "console_scripts": [
+                "spanish-inquisition = montypython.unexpected:spanishinquisition",
+                "brian=montypython.naughtyboy:brian",
+            ],
+            "gui_scripts": [
+                "dead-parrot=montypython.sketch:petshop",
+                "shrubbery=montypython.holygrail:knightswhosayni",
+            ],
+            "project.plugins": [
+                "babysnatchers=montypython.somethingcompletelydifferent:babysnatchers",
+                "eels=montypython.somethingcompletelydifferent:eels",
+            ],
+        }
+    ) == {
+        "console_scripts": {
+            "spanish-inquisition": "montypython.unexpected:spanishinquisition",
+            "brian": "montypython.naughtyboy:brian",
+        },
+        "gui_scripts": {
+            "dead-parrot": "montypython.sketch:petshop",
+            "shrubbery": "montypython.holygrail:knightswhosayni",
+        },
+        "project.plugins": {
+            "babysnatchers": "montypython.somethingcompletelydifferent:babysnatchers",
+            "eels": "montypython.somethingcompletelydifferent:eels",
+        },
+    }
```

Comment:
For longer statements like this, I find it valuable to put the expected result in a variable, and maybe even put the inputs in a variable as well, because when a test fails in pytest, it will print out the failing `assert` statement on the console, and I think it's useful to have that printed statement be short so that it's easy to identify what failed. (Although pytest does abbreviate some things so maybe the output would be reasonable regardless... I guess you could try making some change that would cause this test to fail and check what happens.)

P.S. Nice with the Monty Python references :smile:

Reply:
Yep, was definitely wracking the brain there regarding some references.  It was after all, how the programming language got its name. :-)

I can refactor this, no big deal.